external/fv3fit/fv3fit/_shared/config.py

```diff
@@ -2,15 +2,128 @@
+
+
+class Loadable:
+    @classmethod
+    def load(cls: Type[T], filename: str) -> T:
+        with fsspec.open(filename, "r") as f:
+            config_dict = yaml.safe_load(f)
+        return cls(**config_dict)  # type: ignore
+
+
+@dataclasses.dataclass
+class DataConfig(Dumpable):
+    """Convenience wrapper for model training data
+
+    Attrs:
+        variables: names of variables to include in dataset
+        batch_function: name of function from `fv3fit.batches` to use for
+            loading batched data
+        batch_kwargs: keyword arguments to pass to batch function
+        data_path: location of training data to be loaded by batch function
+        validation_timesteps: timestamps to use as validation samples
+        timesteps_source: one of "timesteps_file",
+            "sampled_outside_input_config", "input_config", "all_mapper_times"
+    """
+
+    variables: List[str]
+    data_path: str
+    batch_function: str
+    batch_kwargs: dict
+
+
 @dataclasses.dataclass
-class ModelTrainingConfig:
+class TrainingConfig(Dumpable):
+    """Convenience wrapper for model training parameters and file info
+
+    Attrs:
+        model_type: sklearn model type or keras model class to initialize
+        hyperparameters: arguments to pass to model class at initialization
+            time
+        input_variables: variables used as features
+        output_variables: variables to predict
+        batch_function: name of function from `fv3fit.batches` to use for
+            loading batched data
+        batch_kwargs: keyword arguments to pass to batch function
+        data_path: location of training data to be loaded by batch function
+        scaler_type: scaler to use for training
+        scaler_kwargs: keyword arguments to pass to scaler initialization
+        additional_variables: list of needed variables which are not inputs
+            or outputs (e.g. pressure thickness if needed for scaling)
+        random_seed: value to use to initialize randomness
+        validation_timesteps: timestamps to use as validation samples
+        save_model_checkpoints: whether to save a copy of the model at
+            each epoch
+        model_path: output location for final model
+        timesteps_source: one of "timesteps_file",
+            "sampled_outside_input_config", "input_config", "all_mapper_times"
+    """
+
+    model_type: str
+    input_variables: List[str]
+    output_variables: List[str]
+    additional_variables: List[str] = dataclasses.field(default_factory=list)
+    hyperparameters: dict = dataclasses.field(default_factory=dict)
+    random_seed: Union[float, int] = 0
+    model_path: str = ""
+
+
+@dataclasses.dataclass
+class KerasTrainingConfig(TrainingConfig):
+
+    save_model_checkpoints: bool = False
+
+
+@dataclasses.dataclass
+class SklearnTrainingConfig(TrainingConfig):
+
+    scaler_type: str = "standard"
+    scaler_kwargs: dict = dataclasses.field(default_factory=dict)
+
+    def __post_init__(self):
```

Comment:
I think it could mislead someone who sees this in the Keras config into thinking keras training supports this, so I'd like to leave it out until we make the support uniform.